integration-cli/docker_api_images_test.go

```diff
@@ -23,59 +24,51 @@ func (s *DockerSuite) TestApiImagesFilter(c *check.C) {
 		v := url.Values{}
 		v.Set("filter", filter)
 		status, b, err := sockRequest("GET", "/images/json?"+v.Encode(), nil)
-		c.Assert(err, check.IsNil)
-		c.Assert(status, check.Equals, http.StatusOK)
+		c.Assert(err, checker.IsNil)
+		c.Assert(status, checker.Equals, http.StatusOK)
 
 		var images []image
-		if err := json.Unmarshal(b, &images); err != nil {
-			c.Fatal(err)
-		}
+		err = json.Unmarshal(b, &images)
+		c.Assert(err, checker.IsNil)
 
 		return images
 	}
 
 	errMsg := "incorrect number of matches returned"
-	if images := getImages("utest*/*"); len(images[0].RepoTags) != 2 {
-		c.Fatal(errMsg)
-	}
-	if images := getImages("utest"); len(images[0].RepoTags) != 1 {
-		c.Fatal(errMsg)
-	}
-	if images := getImages("utest*"); len(images[0].RepoTags) != 1 {
-		c.Fatal(errMsg)
-	}
-	if images := getImages("*5000*/*"); len(images[0].RepoTags) != 1 {
-		c.Fatal(errMsg)
-	}
+	images := getImages("utest*/*")
+	c.Assert(images[0].RepoTags, checker.HasLen, 2)
+
+	images = getImages("utest")
+	c.Assert(images[0].RepoTags, checker.HasLen, 1)
+
+	images = getImages("utest*")
+	c.Assert(images[0].RepoTags, checker.HasLen, 1)
+
+	images = getImages("*5000*/*")
+	c.Assert(images[0].RepoTags, checker.HasLen, 1)
 }
 
 func (s *DockerSuite) TestApiImagesSaveAndLoad(c *check.C) {
 	testRequires(c, Network)
 	testRequires(c, DaemonIsLinux)
 	out, err := buildImage("saveandload", "FROM hello-world\nENV FOO bar", false)
-	if err != nil {
-		c.Fatal(err)
-	}
+	c.Assert(err, checker.IsNil)
 	id := strings.TrimSpace(out)
 
 	res, body, err := sockRequestRaw("GET", "/images/"+id+"/get", nil, "")
-	c.Assert(err, check.IsNil)
-	c.Assert(res.StatusCode, check.Equals, http.StatusOK)
-
+	c.Assert(err, checker.IsNil)
 	defer body.Close()
+	c.Assert(res.StatusCode, checker.Equals, http.StatusOK)
 
 	dockerCmd(c, "rmi", id)
 
 	res, loadBody, err := sockRequestRaw("POST", "/images/load", body, "application/x-tar")
-	c.Assert(err, check.IsNil)
-	c.Assert(res.StatusCode, check.Equals, http.StatusOK)
-
+	c.Assert(err, checker.IsNil)
 	defer loadBody.Close()
```

Comment:
cause you don't need `loadBody` anywhere else. No need to `defer` then, just `.Close`